src/components/SegmentedControl/component.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useCallback } from 'react';
+
+import { Testable, useBuildTestId } from '../../modules/test-ids';
+
+import { Container, Element, Size } from './styled';
+
+export type Props = Omit<React.AllHTMLAttributes<HTMLElement>, 'as' | 'size'> &
+  Testable & {
+    children: React.ReactNode[];
+    selectedIndex?: number;
+    size?: Size;
+  };
+
+export const Component = ({
+  children,
+  onChange,
+  disabled,
+  selectedIndex = 0,
+  'data-testid': testId,
+  size = 'huge',
+  ...otherProps
+}: Props) => {
+  const buildTestId = useBuildTestId(testId);
+  const [selected, setSelected] = useState(children[selectedIndex]);
```

Comment:
You don't need a state here. Just do `children[selectedIndex]` wherever you need to.